### DISCUSSION TITLE
Trouble with spawning after an entity is removed from world

### DISCUSSION THREAD

**Discussion by Harley-L** [AlmasB/FXGL#1134] [Q&A] [answered]
Hello,

This is my first attempt at FXGL and I just ran across an odd error where I can not spawn entities after I remove one from the world. I get the cryptic error message below:

Message:  null cannot be cast to non-null type com.almasb.fxgl.app.scene.GameView
Type:  NullPointerException
Method:  GameScene$sortZ$$inlined$sortBy$1.compare()
Line:  Comparisons.kt:321

Can someone point me in the right direction? My repo: https://github.com/Harley-L/AgeOfWar

Thank you!

**Comment by AlmasB** [comment, marked as answer]
Hi @Harley-L 

I have not encountered one like this before. Based on your `after I remove one from the world`, my guess would be you still have code that tries to do something with the entity that was removed. An entity, once removed from the world, should not be touched in any way.

Hopefully this points you in the right direction.

**Reply by Harley-L** [reply]
Hi @AlmasB

Thanks for the reply. Is there a built-in way I can check if an entity has been removed before I interact with it? Or is the only way to use try-catch blocks?

**Reply by AlmasB** [reply]
`entity.isActive()` - if true, it's in the world, if false it is not.

The lifecycle is:
- isActive is false between creation and adding to the world
- isActive is true when in the world
- isActive is false forever after removing, since entities cannot be readded to the world (well, technically, there is API to make entities reusable, but it is not normally needed for most cases).
